pennylane/templates/subroutines/basis_rotation.py

```diff
@@ -153,35 +154,39 @@ def compute_decomposition(
             list[.Operator]: decomposition of the operator
         """
 
-        M, N = unitary_matrix.shape
+        M, N = qml.math.shape(unitary_matrix)
         if M != N:
             raise ValueError(
                 f"The unitary matrix should be of shape NxN, got {unitary_matrix.shape}"
             )
 
         if check:
-            umat = qml.math.toarray(unitary_matrix)
-            if not np.allclose(umat @ umat.conj().T, np.eye(M, dtype=complex), atol=1e-4):
-                raise ValueError("The provided transformation matrix should be unitary.")
+            umat = qml.math.copy(unitary_matrix)
+            if not qml.math.is_abstract(unitary_matrix):
```

Comment:
Why this check is needed?

Reply:
This prevents the below `unitary` check, which would be incompatible in a JIT-environment.